File: ts/605_Can_Place_Flowers.ts
function canPlaceFlowers(flowerbed: number[], n: number): boolean {
    for (let i = 0; i < flowerbed.length; i++) {
        if (
            flowerbed[i] == 0 
            && (i == 0 || flowerbed[i - 1] == 0)
            && (i == flowerbed.length - 1 || flowerbed[i + 1] == 0)
        ) {
            flowerbed[i] = 1;
            n--;
            i++;
        }
    }

    return n <= 0;
};

File: rust/605_Can_Place_Flowers.rs
impl Solution {
    pub fn can_place_flowers(flowerbed: Vec<i32>, n: i32) -> bool {
        let mut flowerbedClone: Vec<i32> = flowerbed;
        let mut countFlowers: i32 = n;
        for i in 0..flowerbedClone.len() as usize{
            if (
                flowerbedClone[i] == 0 
                && (i == 0 || flowerbedClone[i - 1] == 0)
                && (i == flowerbedClone.len() - 1 || flowerbedClone[i + 1] == 0)
            ) {
                flowerbedClone[i] = 1;
                countFlowers -= 1;

            }
        }  
        return countFlowers <= 0;
    }
}
